Fix failing integration test

Motiejus opened the pull request:
A ping might not even have 'changes' array, add a check for it.

Example:

    ringpop-common/test/reaping-faulty-nodes.js:281
                && ping.body.changes[0].status !== 'alive'  // ignore old ping
                                       ^
    TypeError: Cannot read property 'status' of undefined
        at dsl.validateEventBody._.filter.status (ringpop-common/test/reaping-faulty-nodes.js:281:36)
        at baseFindIndex (ringpop-common/test/node_modules/lodash/index.js:326:11)
        at Function.findIndex (ringpop-common/test/node_modules/lodash/index.js:3261:16)
        at Array.validateEventBody [as 8] (ringpop-common/test/ringpop-assert.js:144:23)
        at progressFromCursor (ringpop-common/test/ringpop-assert.js:844:20)
        at TestCoordinator.<anonymous> (ringpop-common/test/ringpop-assert.js:861:9)
        at TestCoordinator.emit (events.js:117:20)
        at TChannelEndpointHandler.handleRequest (ringpop-common/test/fake-node.js:101:30)
        at RequestCallbackHandler.handleRequest (ringpop-common/test/node_modules/tchannel/request-handler.js:62:23)
        at TChannelEndpointHandler.handleRequest (ringpop-common/test/node_modules/tchannel/endpoint-handler.js:71:17)

mennopruijssers commented on test/reaping-faulty-nodes.js:
```diff
@@ -278,7 +278,7 @@ function pingNotTombstoneState(t, tc) {
             && ping.direction === 'request'
             && ping.receiver !== tc.fakeNodes[0].getHostPort() // ignore nodeIx=0
-            && ping.body.changes[0].status !== 'alive'  // ignore old ping
+            && ! (ping.body.changes[0] && ping.body.changes[0].status === 'alive')
```
did you remove the comment by accident?

Motiejus replied:
Two reasons:

1. The line is explained above already in a separate paragraph -- the comment here predates that explanation.
2. The line length becomes a bit too long for my taste.

mennopruijssers commented:
besides my small question: LGTM

Motiejus commented:
Huh, this change can also fail (happened after executing quite a few times):


```
not ok 3360 The gossip should contain a flagged tombstone
  ---
    operator: ok
    expected: true
    actual:   false
    at: Array.validateEventBody [as 8] (/home/travis/build/uber/ringpop-node/test/ringpop-common/test/ringpop-assert.js:147:11)
  ...
============== error details ===============
{
  "body": {
    "checksum": 2834427358,
    "changes": [],
    "source": "127.0.0.1:27650",
    "sourceIncarnationNumber": 1465204273743
  }
}
```

CorgiMan commented:
From the fn name and comment I get the idea that we don't want a tombstone. From the code it seems that we do want one. I don't get it..

Motiejus commented:
Looks like I opened this PR too early. Sorry, closing.

Motiejus commented:
I ran these two tests for 17 times for all the default cluster sizes each on my local machine for node, and 10 times each for Go, CPU throttled to 10%. [Travis-ci](https://travis-ci.org/uber/ringpop-node/jobs/135568059) seems to be fine with them too.

I also removed `consumePings`, since that's done as part of a separate function inside `validateEventBody`.

@CorgiMan I updated the comments in the code to answer your question.
@mennopruijssers I also changed style a little bit, if you can, please have a look again.

mennopruijssers commented on test/reaping-faulty-nodes.js:
```diff
@@ -274,11 +268,17 @@ test2('tombstone should be gossiped with flag when applied as a flag', getCluste
             && ping.direction === 'request'
             && ping.receiver !== tc.fakeNodes[0].getHostPort() // ignore nodeIx=0
-            && ping.body.changes[0].status !== 'alive'  // ignore old ping
+            && ! isASingleAliveMessage(ping) // race in ringpop-node, described above
```
style nit: remove space after `!`.

mennopruijssers commented:
@Motiejus looks good except one js-style nit.

CorgiMan commented:
Thanks for clarification, Looks good.
